playbook_launch_instance.yml

```diff
@@ -88,4 +89,14 @@
         - "{{ ec2_volumes.results }}"
         - volumes
 
-    - include_tasks: tasks/check_ssh.yml
+    - name: Waiting for SSH to be up
+      wait_for:
+        host: "{{ item.private_ip }}"
+        port: 22
+        delay: 5
+        timeout: 300
+        state: started
+      with_items: "{{ ec2_result.instances }}"
+
+    - pause:
+        seconds: 15
```

Comment:
@mluypaert just wondering if this ssh check should be in the launch_ec2_instance? Then once that task is done the machine is ready for Ansible to start issuing commands against?

Reply:
@oblodgett No, `tasks/launch_ec2_instance.yml` is mainly intended as an "attempt to initiate the EC2 instance" taskset, which gets looped over until one instance initiation succeeds. Once one succeeded, the further tasks in the `playbook_launch_instance.yml` playbook can be executed, non of which rely on SSH being available on the instance. The tasks requiring an SSH connection all get executed in different playbooks, after the import and execution of `playbook_launch_instance.yml`.

FYI, the position of the SSH check task in the execution order has not changed, it only moved to be included in the `playbook_launch_instance.yml` playbook directly, rather than being in a separated taskfile previously.

Reply:
Ahh ok.. sorry didn't look at this close enough.